### PR TITLE
[WIP] Interactivity

### PR DESCRIPTION
**Pull request by Nihlus**
This PR implements interactivity and pagination abstractions for Remora.Discord. At the moment, I'd like community feedback before merging, as it is just a local draft based on code ported from one of my own bot projects.

### CODE REVIEW

**Comment by VelvetThePanda** [comment]
@Nihlus this is a great PR, but due to its size and complexity, I can personally only grasp so much about how it's meant to be used; would you happen to have a sample of the surface of the API/how we'd use it?

File: Remora.Discord.Pagination/PaginatedAppearanceOptions.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    ButtonComponent Close,
+    ButtonComponent Help,
+    string FooterFormat = "Page {0}/{1}",
+    string HelpText = "This is a paginated message. Use the buttons to change page."
+)
+{
+    /// <summary>
+    /// Holds the default appearance instance.
+    /// </summary>
+    public static readonly PaginatedAppearanceOptions Default = new
```
Comment (VelvetThePanda) [inline Remora.Discord.Pagination/PaginatedAppearanceOptions.cs:49]: Curious, why a readonly field over a get-only property?

File: Remora.Discord.Pagination/PaginatedAppearanceOptions.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            ButtonComponentStyle.Secondary,
+            nameof(Last),
+            new PartialEmoji(Name: "⏭"),
+            nameof(Last)
+        ),
+        new ButtonComponent
+        (
+            ButtonComponentStyle.Secondary,
+            nameof(Close),
+            new PartialEmoji(Name: "\x23F9"),
```
Comment (VelvetThePanda) [inline Remora.Discord.Pagination/PaginatedAppearanceOptions.cs:83]: I suppose I also question why an escape code over ⏹

**Review by VelvetThePanda — APPROVED** [review]
Certainly not a thorough review; there's a lot to look through, but it looks fine on the surface thumb

File: Remora.Discord.Pagination/PaginatedAppearanceOptions.cs (hunk 2 above)
Comment (Nihlus) [reply Remora.Discord.Pagination/PaginatedAppearanceOptions.cs:83]: I believe I had a rendering issue with the actual emoji; that might be different now.

File: Remora.Discord.Pagination/PaginatedAppearanceOptions.cs (hunk 1 above)
Comment (Nihlus) [reply Remora.Discord.Pagination/PaginatedAppearanceOptions.cs:49]: No particular reason - I'd be happy changing it over to a property.

**Comment by Nihlus** [comment]
I'll see about adding a sample (honestly thought I already had) :)

File: Remora.Discord.Pagination/PaginatedAppearanceOptions.cs (hunk 2 above)
Comment (VelvetThePanda) [reply Remora.Discord.Pagination/PaginatedAppearanceOptions.cs:83]: Ah the joys of text editors. Had to paste it into a search engine to figure out what emoji it actually was.

**Pull request merged**